modules/OFConnectionManager2/module/src/bundle.c

```diff
@@ -644,7 +654,14 @@ bundle_task(void *cookie)
             subbundle_info.subbundle_idx = state->cur_subbundle;
             subbundle_info.total_msg_count = state->subbundles[state->cur_subbundle].count;
             subbundle_info.is_aborted = state->is_aborted;
-            invoke_subbundle_start(state->cxn_id, &subbundle_info);
+        }
+
+        /* At the end of a subbundle, we need to check whether we
+         * should be paused.
+         */
+        if (cxn && cxn->pending_cnt) {
+            state->cur_subbundle_is_paused = true;
+            return IND_SOC_TASK_CONTINUE;
```

Comment:
not sure why we need all the other code changes to bundle.c; only lines 662 through 664 appear to address the reported issue

Reply:
Try not to invoke the next subbundle start() before all the pending message returned.

Reply:
Try not invoke the start() before the previous subbundle pendings are all returned (no more pending).